### ISSUE TITLE
Remove temp hack once tt-metal support/fix comes via CreateBuffer(allocate=false) for "Metal runtime segfault on close"

### ISSUE THREAD

**Issue by nsmithtt** [tenstorrent/tt-mlir#408]
Sometimes when ttrt finishes executing we segfault when using the metal backend.

```
./build/bin/ttmlir-opt --ttir-load-system-desc="path=n300.ttsys" --ttir-implicit-device --ttir-allocate --convert-ttir-to-ttmetal --ttmetal-serialize-to-binary="output=out.ttm" test/ttmlir/Silicon/TTMetal/to_layout.mlir

ttrt run --identity --atol 1e-02 out.ttm
```

**Comment by kmabeeTT** [comment]
Hmm that's not good, will take a look, thx for raising.

**Comment by nsmithtt** [comment]
It's not a regression, it's been happening this way from day 1. I think there is a buffer that isn't be deallocated or something, at least it's crashing in buffer dealloc. Maybe the runtime map is hanging onto that shared ptr reference? Maybe we need to release the buffer during command queue exec dealloc? I only collected a back trace so I'm jumping to conclusions

**Comment by kmabeeTT** [comment]
TLDR; blocked on Metal [fix](https://github.com/tenstorrent/tt-metal/issues/12046).

Update - I looked at this issue, can reproduce, and I see what's going on.  

- A couple one-line improvements (for visual issues spotted) can be made on tt-mlir side (will put up PR) 
- but it doesn't solve this issue, and we do need a fix on tt-metal side (see https://github.com/tenstorrent/tt-metal/issues/12046 for details, will also put up PR) 

Need the metal feature/bugfix to not allocate and deallocate metal `Buffers` that are created as views for TensorRefs for input/output tensors per program via `createBufferFromTensorRef(device, tensorRef)`, otherwise Buffers that tt-mlir manages (allocating and deallocating) will mistakenly be deallocated by device teardown. 

I made metal changes locally and confirmed it solves this segfault from looking at debug logs, and also from running test in loop 50x, previously would hit segfault 50% of the time.

**Comment by kmabeeTT** [comment]
Update here - TDLR; the "Metal runtimetime segfault on close" is ~worked-around now.

- Merged a PR just now #529 to main which includes those couple one line fixes mentioned in prev comment, and another new tt-mlir hack which works around the original segfault reported here.  Also included is adding the TTMetal existing tests to CI for the first time.  This removes blocking dependency on tt-metal.
- For the tt-metal changes (more proper fix for the segfault, to replace temporary hack), I haven't been able to get passing build inside tt-metal repo (some ttnn cryptic compile errors WIP), so haven't opened PR there quite yet.  Hope to get someone from Metal to look Tuesday if I don't figure it out by then.

Thinking I'll rename this ticket and keep open until proper fix comes.
